scripts/prepdocs.py

```diff
@@ -706,24 +712,25 @@ def read_adls_gen2_files(
         if args.tenantid is None
         else AzureDeveloperCliCredential(tenant_id=args.tenantid, process_timeout=60)
     )
-    default_creds = azd_credential if args.searchkey is None or args.storagekey is None else None
     adls_gen2_creds = azd_credential if args.datalakekey is None else AzureKeyCredential(args.datalakekey)
-    search_creds = default_creds if args.searchkey is None else AzureKeyCredential(args.searchkey)
+    search_creds: Union[TokenCredential, AzureKeyCredential] = azd_credential
```

Comment:
Same issue with ternaries here, had to turn into if/else.

https://github.com/python/mypy/issues?q=is%3Aissue+is%3Aopen+ternary